src/snakemake/rules.py

```
@@ -12,6 +12,7 @@
 from pathlib import Path
 from itertools import chain
 from functools import partial
+from typing import Union
```

Ruff (0.12.2)

`typing.Union` imported but unused

Remove unused import: `typing.Union`

(F401)